spec/services/record_participant_declaration_service_spec.rb

```diff
@@ -2,62 +2,27 @@
 
 require "rails_helper"
 
+require_relative "../shared/context/service_record_declaration_params.rb"
```

Comment:
i don't think i like this for several reasons:
- we're just hiding a load of test setup elsewhere in the code. if someone needs to look into these tests they'd have to look at the shared contexts.
- because the variable names are elsewhere you have to figure out how they affect the current file despite being elsewhere 
- if they want to change any of test setup the will affect every other test setup where it is being called.
- there are 'let!' defined so as soon as the context is called it has affected state
- it loads NPQ and ECF when someone may just want to test one thing

Reply:
Prior to having this setup we were literally copy-pasting the same setup across many different files. It looked much worse than it does now.

That said, I am open to suggestions on how to improve it. Or which of a myriad other solutions for commonising test setup would be better. But the short gist of the problem is that T&P testing scenarios tend to require a bunch of involved setup (schools which are in partnership with LPs and have some teachers for ECF part is probably the worst), and I don't want to have to copy-paste it whenever I need to make a new test suite.

Reply:
I guess, we'll try it out and see what happens. hopefully it'll just work